prog/weaver/main.go

```diff
@@ -575,3 +575,15 @@ func checkFatal(e error) {
 		Log.Fatal(e)
 	}
 }
+
+func filterEmpty(list []string) []string {
+	filtered := list[:0]
```

Comment:
Initialising `filtered` like this will end up modifying the underlying `list` array. Which seems way dodgy to me. https://play.golang.org/p/-wmI6yGLdV